lib/security/JweToken.ts

```diff
@@ -14,6 +15,49 @@ type EncryptDelegate = (data: Buffer, jwk: PublicKey) => Promise<Buffer>;
  * This class hides the JOSE and crypto library dependencies to allow support for additional crypto algorithms.
  */
 export default class JweToken extends JoseToken {
+
+  // used for verification if a JSON Serialized JWS was given
+  private readonly protected: string | undefined;
+  private unprotected: {[key: string]: any} | undefined;
+  private readonly encrypted_key: string | undefined;
+  private readonly iv: string | undefined;
+  private readonly tag: string | undefined;
+  private readonly aad: string | undefined;
+  private readonly isFlattenedJSONSerialized: boolean;
+
+  public constructor (content: string | object, protected cryptoFactory: CryptoFactory) {
+    super(content, cryptoFactory);
+    const jsonContent: any = content;
+    this.isFlattenedJSONSerialized = false;
+    if (typeof jsonContent === 'object' &&
+    'ciphertext' in jsonContent && typeof jsonContent.ciphertext === 'string' &&
+    'iv' in jsonContent && typeof jsonContent.iv === 'string' &&
+    'tag' in jsonContent && typeof jsonContent.tag === 'string' &&
+    (('protected' in jsonContent && typeof jsonContent.protected === 'string') ||
+    ('unprotected' in jsonContent && typeof jsonContent.unprotected === 'object'))) {
+      if ('protected' in jsonContent) {
+        this.protected = jsonContent.protected;
+      }
+      if ('unprotected' in jsonContent) {
+        this.unprotected = jsonContent.unprotected;
+      }
+      this.iv = jsonContent.iv;
+      this.tag = jsonContent.tag;
+      this.aad = jsonContent.aad;
+      if ('recipients' in jsonContent) {
+        // TODO: General JWE JSON Serialization
```

Comment:
Consider creating an issue and reference it for every TODO comment for easy tracking.